### PR TITLE
Migrate to CMake.

### PR DESCRIPTION
**Pull request by jeandudey**
Now the code is organized in components, and uses the ESP-IDF build
system along with CMake.

Now to build the `idf.py build` should be used, and to flash the `idf.py
flash` command.

Notable changes:

1. The `NVS.{h,cpp}` files were renamed to `Storage.{h,cpp}` because it collided with `nvs.h` (on Windows, it's case insensitive)
2. Changed `hwcrypto/sha.h` (it's removed) to `mbedtls/sha1.h`.

Broken things (tested):

- BLE does not work:
```
I (1000) BTDM_INIT: BT controller compile version [340d4d4]
I (1010) system_api: Base MAC address is not set, read default base MAC address from BLK0 of EFUSE
E (1020) BLE_Server: init enable controller failed: ESP_ERR_INVALID_ARG

E (1030) BLE_Server: gatts app register error, error code = 103
I (1030) BLE_Service: create new service
E (1040) BLE_Service: esp_ble_gatts_create_service_failed: ESP_ERR_INVALID_STATE
```

Signed-off-by: Jean Pierre Dudey <jeandudey@hotmail.com>

### CODE REVIEW

**Comment by jeandudey** [comment]
Closes #63

File: components/WS/src/GeneralUtils.cpp [hunk 1]
```diff
@@ -11,7 +11,6 @@
 #include <esp_system.h>
 #include <esp_wifi.h>
 #include <iomanip>
-#include <nvs.h>
 #include <sstream>
 #include <stdio.h>
 #include <string.h>
```
Comment (luisan00) [inline components/WS/src/GeneralUtils.cpp]: Isn't it better to use '<cstdio>' or '<cstring>'?

Comment (jeandudey) [reply components/WS/src/GeneralUtils.cpp]: It's better to use the C++ style headers (<cstdio>), I plan to split this module and remove unused code from it. I plan to do it on another PR after this so others can start working on the code while I refactor that.

Comment (luisan00) [reply components/WS/src/GeneralUtils.cpp]: Ok, this can be changed in another PR, before to complete the milestone

**Pull request merged**